File: articles/loss-framing-and-risk-communication/calculators/r/choice_architecture_calculator.R
#!/usr/bin/env Rscript

# Educational choice architecture calculator.
# Synthetic, self-contained, base R only.

logistic <- function(x) {
  1 / (1 + exp(-x))
}

choice_probability <- function(
  perceived_value = 0.65,
  default_status = 1.0,
  salience = 0.6,
  friction = 0.25,
  trust = 0.7,
  social_norm_signal = 0.45,
  comprehension = 0.8
) {
  z <- -2.15 +
    1.35 * perceived_value +
    0.85 * default_status +
    0.55 * salience -
    1.45 * friction +
    0.80 * trust +
    0.55 * social_norm_signal +
    0.65 * comprehension

  logistic(z)
}

friction_completion_probability <- function(friction = 0.4, sensitivity = 1.2) {
  exp(-sensitivity * friction)
}

with_default <- choice_probability(default_status = 1.0)
without_default <- choice_probability(default_status = 0.0)

summary <- data.frame(
  metric = c(
    "choice_probability",
    "with_default",
    "without_default",
    "default_effect",
    "friction_completion_probability"
  ),
  value = c(
    choice_probability(),
    with_default,
    without_default,
    with_default - without_default,
    friction_completion_probability()
  )
)

args <- commandArgs(trailingOnly = FALSE)
file_arg <- "--file="
script_path <- sub(file_arg, "", args[grep(file_arg, args)])
if (length(script_path) == 0) {
  script_path <- "choice_architecture_calculator.R"
}
script_path <- normalizePath(script_path)
output_dir <- file.path(dirname(dirname(script_path)), "outputs")
if (!dir.exists(output_dir)) {
  dir.create(output_dir, recursive = TRUE)
}

write.csv(summary, file.path(output_dir, "r_choice_summary.csv"), row.names = FALSE)
print(summary)

File: articles/information-disclosure-and-behavioral-response/calculators/python/choice_architecture_calculator.py
#!/usr/bin/env python3
"""
Educational choice architecture calculator.

Synthetic, self-contained, no external dependencies.
"""

from __future__ import annotations

import json
import math
from pathlib import Path
from typing import Dict


def logistic(x: float) -> float:
    return 1.0 / (1.0 + math.exp(-x))


def choice_probability(
    perceived_value: float = 0.65,
    default_status: float = 1.0,
    salience: float = 0.6,
    friction: float = 0.25,
    trust: float = 0.7,
    social_norm_signal: float = 0.45,
    comprehension: float = 0.8,
) -> float:
    """Return a synthetic probability of choosing an option."""
    z = (
        -2.15
        + 1.35 * perceived_value
        + 0.85 * default_status
        + 0.55 * salience
        - 1.45 * friction
        + 0.80 * trust
        + 0.55 * social_norm_signal
        + 0.65 * comprehension
    )
    return logistic(z)


def default_effect() -> Dict[str, float]:
    with_default = choice_probability(default_status=1.0)
    without_default = choice_probability(default_status=0.0)
    return {
        "with_default": with_default,
        "without_default": without_default,
        "default_effect": with_default - without_default,
    }


def friction_completion_probability(friction: float = 0.4, sensitivity: float = 1.2) -> float:
    """Return a synthetic completion probability that declines as friction rises."""
    return math.exp(-sensitivity * friction)


def main() -> None:
    output_dir = Path(__file__).resolve().parents[1] / "outputs"
    output_dir.mkdir(parents=True, exist_ok=True)

    summary = {
        "choice_probability": choice_probability(),
        "default_effect": default_effect(),
        "friction_completion_probability": friction_completion_probability(),
        "ethics_note": "Synthetic educational calculator; evaluate transparency, autonomy, reversibility, and equity before real use.",
    }

    out_path = output_dir / "python_choice_summary.json"
    out_path.write_text(json.dumps(summary, indent=2), encoding="utf-8")
    print(json.dumps(summary, indent=2))


if __name__ == "__main__":
    main()
